obspy/io/sac/util.py

```diff
@@ -462,3 +463,20 @@ def get_sac_reftime(header):
         raise SacHeaderTimeError(msg)
 
     return reftime
+
+
+def _decode_bytes_and_warn(bytestring, key=None, strip_null_terminated=True):
+    """
+    Decodes bytestring to ASCII, ignoring invalid characters if present and
+    issuing a warning if invalid characters are encountered.
+    """
+    if strip_null_terminated:
+        bytestring = re.sub(b'\x00.*', '', bytestring)
```

Comment:
Do you really need `re`? `bytestring.split(b'\x00', 1)[0]` (Also, didn't we have to do this somewhere before?)